src/sandbox/KeywordFinder.java

```diff
@@ -0,0 +1,87 @@
+package sandbox;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class KeywordFinder {
+
+  public static final Logger logger = LogManager.getLogger(KeywordFinder.class);
+  public final List<String> keywords;
+
+  public KeywordFinder(List<String> keywords) {
+    this.keywords = keywords;
+  }
+
+  public List<String> hasKeyword(String text) {
+    List<String> ret = new ArrayList<String>(keywords.size());
+
+    if (keywords != null && text != null) {
+      for (String kw : keywords) {
+        if (text.contains(kw)) {
```

Comment:
Are we accounting for case insensitive matches?  If so, please add checking in a way that handles case insensitivity.